src/gui/folderman.cpp

```diff
@@ -720,22 +720,42 @@ static QString canonicalPath(const QString &path)
     return selFile.canonicalFilePath();
 }
 
-static QString checkPathForSyncRootMarkingRecursive(const QString &path)
+static QString checkPathForSyncRootMarkingRecursive(const QString &path, FolderMan::NewFolderType folderType, const QUuid &accountUuid)
 {
-    auto existingTag = Utility::getDirectorySyncRootMarking(path);
-    if (!existingTag.isEmpty()) {
-        return FolderMan::tr("Folder '%1' is already in use by application %2!").arg(path, existingTag);
+    std::pair<QString, QUuid> existingTags = Utility::getDirectorySyncRootMarkings(path);
+    if (!existingTags.first.isEmpty()) {
+        if (existingTags.first != Theme::instance()->orgDomainName()) {
+            // another application uses this as spaces root folder
+            return FolderMan::tr("Folder '%1' is already in use by application %2!").arg(path, existingTags.first);
+        }
+
+        // Looks good, it's our app, let's check the account tag:
+        switch (folderType) {
+        case FolderMan::NewFolderType::SpacesFolder:
+            if (existingTags.second == accountUuid) {
+                // Nice, that's what we like, the sync root for our account in our app. No error.
+                return {};
+            } else {
+                return FolderMan::tr("Folder '%1' is already in use by another account.").arg(path);
+            }
+
+        case FolderMan::NewFolderType::OC10SyncRoot:
+            [[fallthrough]];
+        case FolderMan::NewFolderType::SpacesSyncRoot:
+            // It's our application but we don't want to create a spaces folder, so it must be another space root
+            return FolderMan::tr("Folder '%1' is already in use by another account.").arg(path);
+        }
```

Comment:
Don't all 3 cases behave the same if the `id != account id`?

With ownCloud 10, the folder would also contain a .syncdb therefor it would be detected as claimed.
WIth ocis, we can only have one sync root, so we can't end up in here a second time.

Anyhow I guess we can combine the error message?

Reply:
done